submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,197 @@
+    // +optional
+    PodSelector PodSelectorSpec `json:"podSelector,omitempty"`
+
+    // Clients can request a specific globalIP to be allocated. If the requested
+    // globalIP is part of globalCIDR of the Cluster and is not allocated to other
+    // K8s objects, then it will be allocated, otherwise creation of the CRD will fail.
+    // Valid values are empty string (""), or a valid IP address from the globalCIDR of
+    // the Cluster. This field cannot be changed through updates.
+    GlobalIP    string          `json:"private_ip"`
+}
+
+type PodSelectorSpec struct {
+    // MatchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
+    // map is equivalent to an element of matchExpressions, whose key field is "key", the
+    // operator is "In", and the values array contains only "value". The requirements are ANDed.
+    MatchLabels map[string]string `json:"matchLabels"`
+
+    // Associated priority to be used for the selected pods.
+    // Its value should be between 0 (lowest) and 1000 (highest) inclusive.
+    // When this value is omitted, the default value of 0 is used.
+    // +optional
+    Priority    int32 `json:"priority,omitempty"`
+}
+```
+
+### User Stories
+
+1.User has two (or more) clusters with Overlapping CIDRs and requires the Pods in one
+  Cluster be able to discover and talk to exported Services.
+
+  To achieve this use-case, user can simply deploy Submariner Globalnet on all the participating
+  clusters. By default, a globalIP will be assigned at the Cluster level which will be used as
+  egressIP for inter-cluster communication and necessary Services can be exported.
+
+2.In a Globalnet deployment user wants the Pods in a particular namespace to use a unique globalIP
+  instead of the globalIP assigned at the cluster level.
+
+  To achieve this, one can apply the following CRD.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalnetEgressIP
+   metadata:
+     name: ns-egressip
+     namespace: myproject
+   spec:
+     globalIP: (optional)
+```
+
+3.In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
+  to use a unique globalIP as egressIP for cross-cluster communication.
+
+  To achieve this, one can apply the following CRD.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalnetEgressIP
+   metadata:
+     name: db-pods
+     namespace: myproject
+   spec:
+     globalIP: (optional)
+     podSelector:
+       matchLabels:
+         role: db
+       priority: 100  
+```
+
+4.In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
+  to use a unique globalIP both as ingress and egressIP for cross-cluster communication.
+
+  To achieve this, user can create a Headless Service (backed by a Deployment or StatefulSets) and
+  Globalnet will ensure that all the backend Pods that belong to the Headless Service are assigned
```

Comment:
Apparently, it doesn't seem to allow users to specify a particular globalIP to a selected Pod (or set of Pods) via `GlobalnetEgressIP` for this case. Can it be specified? Also, does it allow the globalIP not to be changed?

Reply:
Yeah, see the comments in here: https://github.com/submariner-io/enhancements/pull/22/files#r572131272

I'm suggesting that order of priorities to allow what you propose.

Reply:
@mkimuram, as @mangelajo mentioned, we will be updating the priorities for GlobalnetEgressIP (to take highest precedence) and this will enable us to use the same globalIP for egress even when Pods restart. I'll update this PR shortly.